extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -184,17 +184,20 @@ export abstract class AzureAuth implements vscode.Disposable {
 			const currentTime = new Date().getTime() / 1000;
 
 			let accessToken = cachedTokens.accessToken;
+			let expiresOn = Number(cachedTokens.expiresOn);
```

Comment:
everything works as expected when expiresOn is undefined or some other NaN value?

Reply:
Yes we check `typeof expiry === 'number' && !Number.isNaN(expiry)` when checking token expiration. For storing token from auth api response, we always call `Number()` first. `Number(undefined)` is evaluated as `NaN` and won't pass the later check.